src/D2L.Bmx/ConsolePrompter.cs

```diff
@@ -208,16 +144,90 @@ internal class ConsolePrompter : IConsolePrompter {
 		return mfaOptions[index - 1];
 	}
 
-	string IConsolePrompter.GetMfaResponse( string mfaInputPrompt ) {
-		Console.Error.Write( $"{mfaInputPrompt}: " );
-		string? mfaInput = _stdinReader.ReadLine();
+	string IConsolePrompter.GetMfaResponse( string mfaInputPrompt, bool maskInput ) {
+		string? mfaInput;
+
+		if( maskInput ) {
+			mfaInput = GetMaskedInput( $"{mfaInputPrompt}: " );
+		} else {
+			Console.Error.Write( $"{mfaInputPrompt}: " );
+			mfaInput = _stdinReader.ReadLine();
+		}
 
 		if( mfaInput is not null ) {
 			return mfaInput;
 		}
 		throw new BmxException( "Invalid MFA Input" );
 	}
 
+	private string GetMaskedInput( string prompt ) {
+		Func<char> readKey;
+		if( IS_WINDOWS ) {
+			// On Windows, Console.ReadKey calls native console API, and will fail without a console attached
+			if( Console.IsInputRedirected ) {
+				Console.Error.WriteLine( """
+					====== WARNING ======
```

Comment:
It is going to pop this twice once for password + security question, in this case. But meh, that is a very niche edge case